src/soroban/index.ts

```diff
@@ -2,12 +2,13 @@
 /// <reference path="../../types/dom-monkeypatch.d.ts" />
 
 // Expose all types
-export * from './api';
+export * from "./api";
 
 // soroban-client classes to expose
-export { Server, Durability } from './server';
-export { default as AxiosClient } from './axios';
-export { parseRawSimulation, parseRawEvents } from './parsers';
-export * from './transaction';
+export { Server, Durability } from "./server";
+export { default as AxiosClient } from "./axios";
+export { parseRawSimulation, parseRawEvents } from "./parsers";
+export * from "./transaction";
+export * as ContractClient from "./contract_client";
```

Comment:
noted this on Slack, but imo this shouldn't be in SorobanRpc. Coupla options, imo:

* rename `src/soroban/` to `src/soroban/rpc/` and leave it as-is, then move these to the higher level `src/soroban/`
  - you'd still have `StellarSdk.SorobanRpc` but there'd also be `StellarSdk.Soroban` (or, ideally, a better name, e.g. `ContractHelpers` or `Binding` or idk honestly) that contained all or a subset of `src/soroban/contract_client/`
* do this same thing but manually by `delete`ing and injecting stuff into the global export in `src/index.ts`

Reply:
I don't have strong opinions on this. I definitely found it strange to lump this under _RPC_. I thought maybe we already had more Soroban stuff than just RPC stuff lumped into this namespace, so I figured maybe we didn't care. But I guess that's not true.

Could do something like:

- `StellarSdk.Soroban.Rpc`
- `StellarSdk.Soroban.Contract`
- `StellarSdk.Soroban.Contract.Client`
- `StellarSdk.Soroban.Contract.AssembledTransaction`
- etc

Not sure.

Whatever we pick, I want to make sure that we can import the types. Don't want people to need to do what Tyler needed to do with this PR today, using `SorobanRpc.ContractClient.ContractClientOptions` and similar everywhere, because importing those types from `SorobanRpc.ContractClient` didn't work.

Reply:
We could also try using [the `exports` declaration](https://hackmd.io/@dzearing/ry1jTjCKd) to make importing these deeply-nested objects easier.

```ts
import { Client } from 'stellar-sdk/Contract'
import { Api } from 'stellar-sdk/Rpc'
```

Still not sure what the best API is here.

Reply:
Note that in my examples above I dropped `Soroban` from the path. We probably want to do that eventually. Maybe now's the time? Seems like we really just have RPC stuff and ContractClient stuff? I think for the sake of imports it makes sense to import a _Client_ from a path called _Contract_, but I could also see doing `import { Client } from 'stellar-sdk/ContractClient'`